sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1443,6 +1600,151 @@ export class BlobClient extends StorageClient {
     }
   }
 
+  /**
+   * Set the owner, owner group or access control list for a directory.
```

Comment:
As per this comment, the `setAccessControl()` is for the directory.

My apologies, I dont know the details, but I believe ADLS support and the "directory" are related in which case this function is useful after getting the blob client from the directory client by using the `getBlobClient()` method.

Is this function still applicable on the BlobClient gotten by running the `getBlobClient()` method is run on the `ContainerClient`?

Reply:
Same question for the other methods added here like `setPermissions()` and `getAccessControl()`. The comments there point to https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/ 

So am assuming all these are related to ADLS.

Reply:
Yes. Comment should be updated to blob.

 It's a long story about Blob with ADLSgen2 interop. But short version is, both blobClient and directoryClient have 
setAccessControl support when enabled ADLSgen2 interop.



---
In reply to: [335298224](https://github.com/Azure/azure-sdk-for-js/pull/5588#discussion_r335298224) [](ancestors = 335298224)

Reply:
Yes. Will update description to blob. It's for ADLS.

---
In reply to: [335299782](https://github.com/Azure/azure-sdk-for-js/pull/5588#discussion_r335299782) [](ancestors = 335299782)

Reply:
> both blobClient and directoryClient have
setAccessControl support when enabled ADLSgen2 interop.

So, when ADLSgen2 is not enabled, will it not be confusing for the customer to see these 4 methods on the blob client?

Reply:
This API internally calls to DFS endpoint.
ADLS uses the DFS endpoint, Blob uses the blob endpoint.  Both can be used to access and modify the same data 

accountName.dfs.core.windows.net 
accountName.blob.core.windows.net 

In the future, DFS functionality will be added to the blob endpoint, and the DFS endpoint will be deprecated.

So, this API will finally be aviabile for blob endpoint and provided in blob SDK. But for now, it depends on service, for how this API will behavior.

In my testing, this API should work when account HNS enabled. But it may also report "Feature not avaiabile" when HNS not enabled, it depends on interop progress.


---
In reply to: [335592106](https://github.com/Azure/azure-sdk-for-js/pull/5588#discussion_r335592106) [](ancestors = 335592106)